src/main/java/org/sitenv/ccdaparsing/service/CCDAParserAPI.java

```diff
@@ -325,6 +332,15 @@ public CCDARefModel parseCCDA2_1(InputStream inputStream) {
 						isTimeOut = true;
 					}
 				}
+
+				if(familyHistory!=null){
+					try{
+						refModel.setFamilyHistory(familyHistory.get(isTimeOut?minWaitTime:maxWaitTime, TimeUnit.MILLISECONDS));
+					}catch (Exception e) {
+						isTimeOut = true;
+					}
```

Comment:
We have parsed ids in CCDAFamilyHistory, We can add them to the refModels's idList.